src/playlist.js

```diff
@@ -117,7 +117,19 @@ const backwardDuration = function(playlist, endSequence) {
       return { result: result + segment.end, precise: true };
     }
 
-    result += segment.duration;
+    if (segment.preload) {
```

Comment:
This prevents us from considering the full segment duration for preload segments that do not have all of their parts yet.